include/openspace/events/event.h

```diff
@@ -0,0 +1,285 @@
+    static const Type Type = Event::Type::ScreenSpaceRenderableRemoved;
+
+    explicit EventScreenSpaceRenderableRemoved(const ScreenSpaceRenderable* renderable_);
+    const tstring renderable;
+};
+
+
+struct EventCameraTransition : public Event {
+    static const Type Type = Event::Type::CameraTransition;
+
+    enum class Location {
+        Outside,
+        ApproachSphere,
+        ReachedSphere
+    };
+    EventCameraTransition(const Camera* camera_, const SceneGraphNode* node_,
+        Location before_, Location after_);
+
+    const Camera* camera = nullptr;
+    const tstring node;
+    const Location before;
+    const Location after;
+};
+
+
+// Not used right now
+struct EventTimeOfInterestReached : public Event {
+    static const Type Type = Event::Type::TimeOfInterestReached;
+
+    EventTimeOfInterestReached(const Time* time_, const Camera* camera_);
+    const Time* time = nullptr;
+    const Camera* camera = nullptr;
+};
+
+
+// Not used right now
+struct EventMissionEventReached : public Event {
+    static const Type Type = Event::Type::MissionEventReached;
+
+    // Not sure which kind of parameters we want to pass here
+    EventMissionEventReached();
+};
+
+
+// Not used right now
+struct EventPlanetEclipsed : public Event {
+    static const Type Type = Event::Type::PlanetEclipsed;
+
+    EventPlanetEclipsed(const SceneGraphNode* eclipsee_, const SceneGraphNode* eclipser_);
+    const tstring eclipsee;
+    const tstring eclipser;
+};
+
+
+struct EventInterpolationFinished : public Event {
+    static const Type Type = Event::Type::InterpolationFinished;
+
+    EventInterpolationFinished(const properties::Property* property_);
+    const tstring property;
+};
+
+
+struct EventFocusNodeChanged : public Event {
+    static const Type Type = Event::Type::FocusNodeChanged;
+
+    EventFocusNodeChanged(const SceneGraphNode* oldNode_, const SceneGraphNode* newNode_);
+    const tstring oldNode;
+    const tstring newNode;
+};
+
+
+struct EventLayerAdded : public Event {
+    static const Type Type = Event::Type::LayerAdded;
+
+    explicit EventLayerAdded(std::string_view node_, std::string_view layerGroup_, 
+        std::string_view layer_);
+    const tstring node;
+    const tstring layerGroup;
+    const tstring layer;
+};
+
+
+struct EventLayerRemoved : public Event {
+    static const Type Type = Event::Type::LayerRemoved;
+
+    explicit EventLayerRemoved(std::string_view node_, std::string_view layerGroup_, 
+        std::string_view layer_);
+    const tstring node;
+    const tstring layerGroup;
+    const tstring layer;
+};
+
+
+struct EventSessionRecordingPlayback : public Event {
+    static const Type Type = Event::Type::SessionRecordingPlayback;
+
+    enum class State {
+        Started,
+        Paused,
+        Resumed,
+        Finished
+    };
+
+    EventSessionRecordingPlayback(State state_);
+    const State state;
+};
+
+
+struct CustomEvent : public Event {
```

Comment:
Add a comment about the intention about this event type and its intended use case (and especially, when it should not be used)